include/pool_alloc.h

```diff
@@ -0,0 +1,34 @@
+#include <queue.h>
+#include <vm.h>
+#include <bitstring.h>
+
+typedef struct pool_slab {
+  LIST_ENTRY(pool_slab) ph_slablist; /* pool slab list */
+  vm_page_t *ph_page;                /* page containing this slab*/
+  uint16_t ph_nused;                 /* # of items in use */
+  uint16_t ph_nfree;  /* # of free (and available) items, a bit redundant but
+                         there would be padding abyway*/
+  uint16_t ph_ntotal; /* total number of chunks*/
+  uint16_t ph_start;  /* start offset in page */
+  bitstr_t ph_bitmap[0];
+} pool_slab_t;
```

Comment:
The definition of the contents of this structure is only needed inside `pool_alloc.c`. Moving the structure definition there will make it invisible from other files, which will clean up the interface a bit, and will prevent people from accessing slab metadata.